identity_account/src/error.rs

```diff
@@ -36,6 +36,12 @@ pub enum Error {
   /// Caused by verification methods without fragments.
   #[error("method missing fragment")]
   MethodMissingFragment,
+  /// Caused by reaching an invalid state for the identity.
+  #[error("invalid identity state: {0}")]
+  InvalidIdentityState(String),
+  /// Caused by an error when serialing or deserialing IdentityState.
+  #[error("serialization error: {0}")]
+  SerializationError(String, #[source] serde_json::Error),
```

Comment:
We could probably get away with a single `InvalidBlob` error variant with a string, and convert the inner error to a string as-needed? Not sure how if that aligns with our error handling conventions though. Removing `serde_json` and relying on `ToJson` will make the errors `CoreError` by the way, if that makes a difference.

@PhilippGackstatter?

Reply:
Using `CoreError` seems fine. These are errors that a user cannot cause, so debuggability isn't a priority. So we don't really need a dedicated variant in my opinion.
A similar argument applies to `InvalidIdentityState`, but it's less clear what to replace it with, so I'm fine with keeping it.